kernel/aster-nix/src/process/process_vm/init_stack/mod.rs

```diff
@@ -111,11 +111,25 @@ pub struct InitStack {
 
 impl Clone for InitStack {
     fn clone(&self) -> Self {
-        Self {
-            initial_top: self.initial_top,
-            max_size: self.max_size,
-            pos: self.pos.clone(),
-            vmo: self.vmo.dup(),
+        if self.is_initialized() {
+            // `InitStack` only use the VMO to do initialization.
+            // The VMO should be empty in the cloned InitStack, thus the `InitStack` can init
+            // with new `argv` and new `envp` again.
+            let vmo = {
+                let vmo_options = VmoOptions::<Full>::new(self.max_size);
+                vmo_options.alloc().unwrap()
```

Comment:
The semantics of cloning is to have an identical copy of the original object. But this clone method does not conform to this standard semantics. If the user of `InitStack::clone` needs an empty stack, then they should not have called the `clone` method.

Even if the user should call `clone`, this body of this method shares most its code with the `new` method. We should avoid duplicated code.